azure-pipelines/release.yml

```diff
@@ -15,66 +15,100 @@ parameters:
 variables:
 # This is expected to provide VisualStudioMarketplacePAT to the release (https://code.visualstudio.com/api/working-with-extensions/publishing-extension#get-a-personal-access-token)
 - group: vscode-csharp release secrets
+# This is expected to provide pat to tag release.
+- group: DncEng-Partners-Tokens
 
-jobs:
-- job: PublishToMarketplace
-  pool:
-    vmImage: ubuntu-latest
-  steps:
-  - task: DownloadPipelineArtifact@2
-    displayName: '📦 Download artifacts from build pipeline.'
-    inputs:
-      buildType: 'specific'
-      project: 'internal'
-      definition: 1264
-      buildVersionToDownload: 'specific'
-      buildId: '$(resources.pipeline.officialBuildCI.runID)'
-      branchName: '$(resources.pipeline.officialBuildCI.sourceBranch)'
-  - pwsh: |
-      npm install --global vsce
-    displayName: 'Install vsce'
-  - pwsh: |
-      # Our build pipeline would generated build based on attempt number. Publishing the latest attempt.
-      $allArtifacts = Get-ChildItem -Path "VSIXs - Attempt*" | Sort-Object -Descending
-      if ($allArtifacts.Length -eq 0) {
-        throw "No Artifacts is downloaded."
-      }
+stages:
+- stage: PublishStage
+  jobs:
+  - job: PublishToMarketplace
```

Comment:
Apologies I just merged my approval change which created merge conflicts.  This will need to take the change from my PR to turn this into a deployment job - https://github.com/dotnet/vscode-csharp/pull/6139